frontend/src/components/game/PlayerGameView.tsx

```diff
@@ -210,6 +262,13 @@ function PlayerGameView(props: PlayerGameViewProps) {
   // Map the game in Redux to the state variables used in this file
   useEffect(() => {
     if (game && currentUser && currentUser.userId) {
+      // Get the new player object for spectating.
+      game.players.forEach((player) => {
+        if (player.user.userId === spectateGame?.user.userId) {
+          setPlayerSpectate(player);
+        }
```

Comment:
At first I was kinda confused by this and other parts of code due to the variable naming - any chance we could do a quick refactor to maybe clear it up a bit? These variables are kinda tough to think of names for, but maybe something like spectatedPlayer or currentlySpectatedPlayer? 

Along a similar line, SpectateGame initially made me think of an action as opposed to like SpectateGameInfo or GameSpectateData - I missed this in the first PR though so nbd, but if possible could you add a comment for that type to describe that it's the info sent via the socket to allow the live previewing by another user?

Reply:
Sure, I cn clear up that variable naming and add some comments.

Reply:
Done! ✅